packages/core/src/editor/Block.css

```diff
@@ -184,6 +184,24 @@ NESTED BLOCKS
   gap: 1.2em;
 }
 
+/* Checked */
```

Comment:
makes me think, would it even be possible to use a label for the inline content + checkbox input? That would be the most "semantic" / accessible way, but not sure if that only introduces more complexity in terms of keyboard handling etc

Reply:
Tested this and while keyboard navigation is no problem, clicking the text (which is now a label) toggles the checkbox, which is default browser behaviour so I don't think there's an easy workaround (assuming the checkbox `id` and label `for` attributes match). Also not sure if the semantics are correct actually, since the `contentDOM` is more like a second input field rather than a label.

Reply:
@matthewlipski you can check how Tiptap handles that in their own extension: https://tiptap.dev/docs/editor/api/nodes/task-item

Clicking the text shouldn't trigger the checkbox, as this can become annoying for the user

Reply:
@matthewlipski Is there any way we can use the mantine checkbox with the indigo colorway. I think it looks a lot better than the one in the current preview. Would really appreciate this change. I am currently using a custom schema block that does the todo list with the mantine checkbox item. Here's what it looks like:

<img width="1440" alt="Screenshot 2024-05-14 at 10 29 29 PM" src="https://github.com/TypeCellOS/BlockNote/assets/53831852/c476502d-eb14-4457-8ad4-db459a4772ca">

Reply:
I'm also happy to paste my custom code even though most if it is a crummy fix to the block not continuing on a new line. 



`'use client';
import { defaultProps, insertOrUpdateBlock } from '@blocknote/core';
import { createReactBlockSpec } from '@blocknote/react';
import { Box, Checkbox, Flex } from '@mantine/core';
import { IconCheck } from '@tabler/icons-react';
import { useEffect } from 'react';
import CategorySelect from './CategorySelect';
import './styles.css';

// The types of alerts that users can choose from.

export const insertCheckBox = (editor) => ({
  title: 'Todo',
  onItemClick: () => {
    insertOrUpdateBlock(editor, {
      type: 'checkbox',
    });
  },
  aliases: ['checkbox', 'todo', 'list', 'bullets', 'to-do', 'tasks', 'taskboard', 'board'],
  subtext: 'Plan out your tasks using a checkbox todo list',
  //   group: 'Other',
  icon: <IconCheck size="20" />,
});
// The Alert block.
export const CheckBoxBlock = createReactBlockSpec(
  {
    type: 'checkbox',
    propSchema: {
      textAlignment: defaultProps.textAlignment,
      textColor: defaultProps.textColor,
      type: {
        default: false,
        values: [true, false],
      },
    },
    content: 'inline',
  },
  {
    render: (props) => {
      useEffect(() => {
        //LOGIC FOR NEW CHECKBOX ON ENTER
        // console.log('hello');
        const handleKeyDown = (event) => {
          if (event.key == 'Tab') {
            event.preventDefault();
            return;
          }
          if (event.key === 'Enter') {
            if (props?.editor?.getTextCursorPosition()?.prevBlock?.id != props.block.id) {
              return;
            }

            //!OLD STUFF BELOW
            var thisFunctionalComponentBlock = props.editor.getTextCursorPosition().prevBlock;

            if (thisFunctionalComponentBlock.content.length == 0) {
              props.editor.updateBlock(props.editor.getTextCursorPosition().prevBlock, {
                type: 'paragraph',
                // props: { type: event.currentTarget.checked },
              });

              props.editor.setTextCursorPosition(
                props.editor.getTextCursorPosition().prevBlock,
                'start'
              );
              if (props.editor.getTextCursorPosition().nextBlock.content.length == 0) {
                props.editor.removeBlocks([props.editor.getTextCursorPosition().nextBlock]);
              }
              return;
            }

            props.editor.updateBlock(props.editor.getTextCursorPosition().block, {
              type: 'checkbox',
              // props: { type: event.currentTarget.checked },
            });
          }
        };

        document.addEventListener('keydown', handleKeyDown);

        return () => {
          document.removeEventListener('keydown', handleKeyDown);
        };
      }, []); // empty dependency array to run effect only once

      return (
        <Flex
          // align="center"

          style={{ textWrap: 'wrap', wordBreak: 'break-word' }}
          gap="xs"
          data-alert-type={props.block.props.type}
        >
          {/* <div style={{ textWrap: 'wrap', wordBreak: 'break-word' }} contentEditable={false}> */}
          <div contentEditable={false}>
            <Checkbox
              checked={props.block.props.type}
              // value={props.block.props.type}
              mt="2"
              ml="2"
              onChange={(event) =>
                props.editor.updateBlock(props.block, {
                  type: 'checkbox',
                  props: { type: event.currentTarget.checked },
                })
              }
              radius="1"
              color="indigo"
            />
          </div>

          {/* <Icon
              className={'alert-icon'}
              data-alert-icon-type={props.block.props.type}
              size={32}
            /> */}
          {/* </div> */}
          {/*Rich text field for user to type in*/}
          <Box
            mt="1"
            className={'inline-content'}
            style={{ textDecoration: props.block.props.type ? 'line-through' : '' }}
            ref={props.contentRef}
          />

          {/* <Box contentEditable={false} ml="auto" mr="0">
            <CategorySelect />
          </Box> */}
        </Flex>
      );
    },
  }
);
`